### PR TITLE
Handle termination HTTP signal

### PR DESCRIPTION
**Pull request by amcintosh**
Provide a method of remotely terminating the proxy.

See Issue #77

### CODE REVIEW

**Comment by abutaha** [comment]
Hi,

Cool feature! Will have it merged soon.

Thanks,

**Comment by amcintosh** [comment]
Thanks!

File: CONTRIBUTORS.md [hunk 1]
```diff
@@ -24,4 +24,5 @@ https://github.com/diranged
 https://github.com/em0ney
 https://github.com/zqben402
 https://github.com/dlackty
+https://github.com/amcintosh
```
Comment (minichate) [inline CONTRIBUTORS.md:27]: 😻

File: aws-es-proxy.go [hunk 2]
```diff
@@ -210,6 +212,10 @@ func (p *proxy) getSigner() *v4.Signer {
 }
 
 func (p *proxy) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	if p.remoteTerminate && r.URL.Path == "/terminate-proxy" && r.Method == http.MethodPost {
+		logrus.Infoln("Terminate Signal")
+		os.Exit(0)
```
Comment (minichate) [inline aws-es-proxy.go:217]: Amazing! ❤️

File: aws-es-proxy.go [hunk 3]
```diff
@@ -210,6 +212,10 @@ func (p *proxy) getSigner() *v4.Signer {
 }
 
 func (p *proxy) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	if p.remoteTerminate && r.URL.Path == "/terminate-proxy" && r.Method == http.MethodPost {
```
Comment (minichate) [inline aws-es-proxy.go:215]: How did you land on using `POST`?

Comment (amcintosh) [reply aws-es-proxy.go:215]: I didn't want to terminate on a GET and while I considered DELETE, this isn't exactly restful, so I figured POST to be the most intuitive.

**Comment by amcintosh** [comment]
Hi @abutaha. Do you know when you'll be able to merge this? Thanks!

**Review by minichate — APPROVED** [review]

**Review by phntom — APPROVED** [review]
i would like this change merged. @amcintosh do you have a docker hub fork of this built?

**Comment by amcintosh** [comment]
@phntom I do not at the moment.

**Comment by abutaha** [comment]
Apologies for the delay. I'll merge this ASAP.

**Comment by phntom** [comment]
i've deployed your branch under phntom/aws-es-proxy:0.9.1 but i'll switch when an official release is available, seems to be working well with python requests.post to the url.

**Comment by enver** [comment]
@abutaha Any chance you get this released?

**Pull request merged**